mk/spksrc.service.mk

```diff
@@ -131,10 +131,15 @@ ifeq ($(STARTABLE),no)
 $(DSM_SCRIPTS_DIR)/start-stop-status: $(SPKSRC_MK)spksrc.service.non-startable
 	@$(dsm_script_copy)
 else
+ifneq ($(strip $(SERVICE_EXE)),)
```

Comment:
So this only works if `SERVICE_EXE` is defined in the package's `Makefile`. If it's in `service-setup.sh` then this variable is empty since that file and the variables not available when compiling.

Similarly when I just started Generic Service it took me a longgg time to understand why I kept getting `Set STARTABLE=no or provide either SERVICE_COMMAND or specific SSS_SCRIPT`, until I realized I was forced to put `STARTABLE=yes` in the `Makefile` for it to skip this error.

Is that the intended way or a design-flaw?

Reply:
You're right, this error message should not be thrown. I will fix in my WIP PR.

As `SERVICE_EXE` is supposed to be absolute path to executable, I do not think it is relevant to set it in `service-setup.sh`. You have multiple options: either use symbolic links like `/var/packages/PACKAGE/bin/exe` or `/usr/local/PACKAGE/bin/exe`. If you prefer to "generate" it from package location, this should work too: `SERVICE_EXE=$${SYNOPKG_PKGDEST}/bin/exe`

Reply:
Aah, so don't want to keep it all in `service-setup.sh`? Since so far we have been doing that for all packages. For me it's all good, just wanted to know 😉

Reply:
Yes I know... but I have to design some kind of "API" so that build chain is able to select "start-stop-status" base script... If you have a better proposal, do not hesitate.

Reply:
How about a `SERVICE_DAEMON=1` for the `Makefile` and then keep the rest in the `service-setup.sh`?

Reply:
In that regard, `STARTABLE` in `Makefile` maybe isn't that bad after all!

Reply:
A single switch like `SERVICE_DAEMON` will not inform `Makefile` about which `start-stop-status` script to use...  `mk/spksrc.service.start-stop-daemon` or generic `mk/spksrc.service.start-stop-status` or maybe/probably yet another future alternative. I will think about it and propose something here